src/pages/edit_resource/contributors_step/CreatorField.tsx

```diff
@@ -119,10 +124,16 @@ const CreatorFields: FC<CreatorFieldsProps> = ({ setAllChangesSaved }) => {
   };
 
   const calculateNumSMColumns = (index: number) => {
-    if (index === 0 && values.creators?.length < 2) {
+    if (index === 0 && values.creators?.length < 2 && user.institutionAuthorities?.isCurator) {
+      return 7;
+    } else if (index === 0 && values.creators?.length >= 2 && user.institutionAuthorities?.isCurator) {
+      return 5;
+    } else if (index === 0 && values.creators?.length < 2 && !user.institutionAuthorities?.isCurator) {
       return 10;
-    } else if (index === 0 && values.creators?.length >= 2) {
+    } else if (index === 0 && values.creators?.length >= 2 && !user.institutionAuthorities?.isCurator) {
       return 8;
+    } else if (user.institutionAuthorities?.isCurator) {
+      return 6;
     } else {
       return 9;
     }
```

Comment:
Syntes denne greia her så enda værre ut da jeg nøstet if setningene. Hva synes du?

Reply:
Ser oversiktelig ut, er uansett vanskelig å lese. Hadde det ikke vært enklere å styre stylingen selv med flex grow og media-wrappere ?

Reply:
Hva med evt å lage variabler for showAutoritySelector etc...- det kunne vært enklere å lese intensjonen bak koden. Nå er det ganske tungrodd å tolke hvorfor f.ex antall kolonner skal bli 8 for` index === 0 && values.creators?.length >= 2 && !user.institutionAuthorities?.isCurator`
Tror også om vi hadde flyttet ut hjelpe-knappen til headeren - hadde logikken blitt enkelere.

Reply:
Endret til egen lagde flex-boxer istedet for grid